python/cloudtik/core/_private/cluster/cluster_operator.py

```diff
@@ -1172,6 +1173,13 @@ def _set_up_config_for_head_node(config: Dict[str, Any],
             remote_key_path: config["auth"]["ssh_private_key"],
         })
 
+    # rsync ssh_public_key to head node authorized_keys,
+    # it's for kubernetes to use.
+    if not is_use_internal_ip(config) and config["provider"]["type"] == "kubernetes":
```

Comment:
The same issue, we need to avoid "type" == "kubernetes" and do something.
And can we handle this in Kubernetes provider?